contracts/receivers/aave-v2-receiver/main.sol

```diff
@@ -126,6 +126,50 @@ contract AaveV2Migrator is MigrateResolver {
     }
 }
 
+contract InstaFlash is AaveV2Migrator {
+
+    using SafeERC20 for IERC20;
+
+    /**
+     * FOR SECURITY PURPOSE
+     * only Smart DEFI Account can access the liquidity pool contract
+     */
+    modifier isDSA {
+        uint64 id = instaList.accountID(msg.sender);
+        require(id != 0, "not-dsa-id");
+        _;
+    }
+
+    function initiateFlashLoan(
+        address[] calldata _tokens,	
+        uint256[] calldata _amounts,	
+        uint _route, // no use of route but just to follow current flashloan pattern
+        bytes calldata data
+    ) external isDSA {	
+        uint _length = _tokens.length;
+        require(_length == _amounts.length, "not-equal-length");
+        uint[] memory iniBal = new uint[](_length);
+        IERC20[] memory _tokenContracts = new IERC20[](_length);
+        for (uint i = 0; i < _length; i++) {
+            _tokenContracts[i] = IERC20(_tokens[i]);
+            iniBal[i] = _tokenContracts[i].balanceOf(address(this));
+            _tokenContracts[i].safeTransfer(msg.sender, _amounts[i]);
```

Comment:
What if we send `0xeeee...eee`? Should we convert it into wmatic address in the connector?